plugins/connection/docker.py

```diff
@@ -308,11 +330,32 @@ def put_file(self, in_path, out_path):
                 count = ' count=0'
             else:
                 count = ''
-            args = self._build_exec_cmd([self._play_context.executable, "-c", "dd of=%s bs=%s%s" % (out_path, BUFSIZE, count)])
+            if getattr(self._shell, "_IS_WINDOWS", False):
+                cmd = '''& { $strm = [Console]::OpenStandardInput(1024)
+$data = [System.Array]::CreateInstance([byte],1024)
+$fstrm = [IO.File]::OpenWrite(%s)
+do {
+    $read = $strm.Read($data, 0, $data.Length)
+    if ($read -gt 0) {
+        $fstrm.Write($data, 0, $read)
+    }
+} while ($read -gt 0)
+$fstrm.Flush()
+$fstrm.Dispose()
+$strm.Dispose() }''' % (out_path)
+                args = self._build_exec_cmd(["cmd", "/c", self._shell._encode_script(cmd, as_list=False, strict_mode=False, preserve_rc=False)])
```

Comment:
This will start `cmd /c` to then call `powershell`. You may as well just start `powershell` direct to make it more efficient

```suggestion
                args = self._build_exec_cmd(self._shell._encode_script(cmd, as_list=True, strict_mode=False, preserve_rc=False))
```